[lg-1] Add file appender

OleksandrGubariev opened the pull request:
Configure file appender for logback.
1. All logs, starting from level DEBUG in your application, should be stored in file.
2. For file name, use next pattern:
- current log: photostudio.log
- previous logs: photostudio-[yyyy-MM-dd].log[.log_number], for example:
photostudio-2017-10-15.log.0 - for first log file which was published on 2017-10-15
3. Maximum size of log file is 5mb.

OleksandrGubariev committed:
1.Add logs to all classes
2.Add file logback.xml

Brutt commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -55,7 +61,7 @@ public JdbcOrderDao(DataSource dataSource) {
 
             String selectOrdersByParameters = addSort(GET_ALL_ORDERS + resultWhere);
-
+            LOG.info("Get orders by parameters from DB");
```
Place this row at the very beginning of the method. Here add debug with incoming parameters.

Brutt commented on src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java:
```diff
@@ -90,15 +98,17 @@ public void add(User user) {
 
     @Override
     public User getUserById(long id) {
+        LOG.info("Get user by id");
```
Add incoming param to log

Brutt commented on src/main/java/com/photostudio/security/impl/DefaultSecurityService.java:
```diff
@@ -7,31 +7,39 @@
     @Override
     public Session login(String login, String password) {
+        LOG.info("Connect user by login: {}", login);
+        LOG.debug("Connect user by login: {} {}", login, password);
```
remove password from logs

viktorterljuk replied:
Are you sure you need both of these? The better message, in this case, will be something like "Trying to log in the user with login: ... and password: ...".

Brutt commented on src/main/java/com/photostudio/service/impl/DefaultUserService.java:
```diff
@@ -15,31 +18,38 @@ public DefaultUserService(UserDao userDao) {
     public List<User> getAllUsers() {
-        return userDao.getAllUsers();
+        List<User> allUsers = userDao.getAllUsers();
+        LOG.info("Started service get all users from DB: {}", allUsers);
```
change to Debug/Trace

Brutt commented on src/main/java/com/photostudio/util/PropertyReader.java:
```diff
@@ -14,22 +18,25 @@ public PropertyReader(String path) {
     public Properties getProperties() {
         String prodEnvironment = System.getenv("environment");
-
+        LOG.info("System environment: {}", prodEnvironment);
```
no need this log

Brutt commented on src/main/java/com/photostudio/util/PropertyReader.java:
```diff
@@ -14,22 +18,25 @@ public PropertyReader(String path) {
-
+            String dbUrl = System.getenv("JDBC_DATABASE_URL");
+            properties.setProperty("jdbc.url", dbUrl);
+            LOG.info("Set jdbc.url: {}", dbUrl);
```
change to debug

Brutt commented on src/main/java/com/photostudio/util/PropertyReader.java:
```diff
@@ -40,9 +47,11 @@ private Properties getDevProperties() {
                 throw new IllegalArgumentException("No properties on path " + path);
             }
             properties.load(inputStream);
+            LOG.info("Read properties from path: {}", path);
```
change to debug

Brutt commented on src/main/java/com/photostudio/web/servlet/user/UserServlet.java:
```diff
@@ -43,6 +48,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        LOG.info("Request with user data for registration user received");
```
Think about to log all params with trace

viktorterljuk commented on src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java:
```diff
@@ -1,7 +1,14 @@
     public LoginPasswordInvalidException(String message) {
+
         super(message);
+        LOG.info("Run LoginPasswordInvalidException with message: {}", message);
```
I am not sure that logging inside an exception is a good idea. When it occurs you'll definitely know it without a log message.

Rooman replied:
This is hyevaya idea.

viktorterljuk commented on src/main/java/com/photostudio/web/listener/ThymeleafInitializerListener.java:
```diff
@@ -1,15 +1,19 @@
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
     @Override
     public void contextInitialized(ServletContextEvent sce) {
         System.out.println("Initializing thymeleaf processor");
```
As logging added, you don't need this line anymore.

ValentynaBondarenko commented on src/main/java/com/photostudio/security/impl/DefaultSecurityService.java:
```diff
@@ -7,31 +7,39 @@
             Session session = Session.builder().user(user)
                     .token(userToken).expireDate(LocalDateTime.now().plusHours(2)).build();
             sessionList.add(session);
+            LOG.info("User with login: {} is logged in ", login);
```
should there be LOG.info or LOG.debug?

OleksandrGubariev committed:
1.JdbcOrderDao: Place log.info at the very beginning of the method. And add debug with incoming parameters.
2.JdbcUserDao: Add incoming param to log.
LoginPasswordInvalidException: delete log.
4.DefaultSecurityService: delete password from logs
5.DeefaultUserService:change info to debug
6.PropertyReader: delete log.info
7.ThymeleafInitializerListener: delete System.out.println()
8.UserServlet: add log.debug with all params

Brutt commented on src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java:
```diff
@@ -68,14 +73,18 @@ public JdbcUserDao(DataSource dataSource) {
         } catch (SQLException e) {
-            throw new RuntimeException("Can't show all users", e);
+            LOG.error("An exception occurred while trying to get all users", e);
+            throw new RuntimeException("Can't show all users");
```
backthrow new RuntimeException("Can't show all users"**, e**);

Brutt commented on src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java:
```diff
@@ -110,10 +122,13 @@ public User getUserById(long id) {
         } catch (SQLException e) {
-            throw new RuntimeException("Can't found user where id=" + id, e);
+            LOG.error("An exception occurred while trying to get user by id: {}", id, e);
+            throw new RuntimeException("Can't found user where id=" + id);
```
throw new RuntimeException("Can't found user where id=" + id, **e**);

Brutt commented on src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java:
```diff
@@ -123,17 +138,21 @@ public void edit(User user) {
         } catch (SQLException e) {
-            throw new RuntimeException("Can't remove user", e);
+            LOG.error("An exception occurred while trying to delete user by id: {}", id, e);
+            throw new RuntimeException("Can't remove user");
```
throw new RuntimeException("Can't remove user", **e**);

Brutt commented on src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java:
```diff
@@ -145,16 +164,21 @@ public User getByLogin(String login) {
         } catch (SQLException e) {
-            throw new RuntimeException("Get user by login error", e);
+            LOG.error("An exception occurred while trying to get user with login: {}", login, e);
+            throw new RuntimeException("Get user by login error");
```
throw new RuntimeException("Get user by login error", **e**);

Brutt commented on src/main/java/com/photostudio/exception/LoginPasswordInvalidException.java:
```diff
@@ -1,6 +1,11 @@
+import org.slf4j.LoggerFactory;
+
 public class LoginPasswordInvalidException extends RuntimeException {
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
```
no need it

Brutt commented on src/main/java/com/photostudio/security/impl/DefaultSecurityService.java:
```diff
@@ -7,31 +7,38 @@
             String actualPassword = user.getPasswordHash();
             if (!hashedPassword.equals(actualPassword)) {
-                throw new LoginPasswordInvalidException("Login/password invalid for user" + login);
+                LOG.error("Password invalid for user: {}", login);
```
This message reveals the true cause that **password** is invalid. Change to "Login/password invalid for user"

Brutt commented on src/main/java/com/photostudio/service/impl/DefaultUserService.java:
```diff
@@ -15,31 +18,38 @@ public DefaultUserService(UserDao userDao) {
     public List<User> getAllUsers() {
-        return userDao.getAllUsers();
+        List<User> allUsers = userDao.getAllUsers();
+        LOG.debug("Started service get all users from DB: {}", allUsers);
```
Why do you use info in DefaultOrderService and here use debug?

Brutt commented on src/main/java/com/photostudio/util/PropertyReader.java:
```diff
@@ -40,9 +47,11 @@ private Properties getDevProperties() {
         } catch (IOException e) {
-            throw new RuntimeException("Can't read properties file " + path, e);
+            LOG.error("Can't read properties file: {} ", path, e);
+            throw new RuntimeException("Can't read properties file " + path);
```
throw new RuntimeException("Can't read properties file " + path, **e**);

Brutt commented on src/main/java/com/photostudio/web/servlet/AdminPageServlet.java:
```diff
@@ -11,18 +13,20 @@
         } catch (IOException e) {
-            throw new RuntimeException("AdminPageServlet error", e);
+            LOG.error("AdminPageServlet error", e);
+            throw new RuntimeException("AdminPageServlet error");
```
throw new RuntimeException("AdminPageServlet error", **e**);

Brutt commented on src/main/java/com/photostudio/web/servlet/LoginServlet.java:
```diff
@@ -50,7 +55,8 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         } catch (IOException e) {
-            throw new RuntimeException("LoginServlet doPost error", e);
+            LOG.error("LoginServlet doPost error", e);
+            throw new RuntimeException("LoginServlet doPost error");
```
throw new RuntimeException("LoginServlet doPost error", **e**);

Brutt commented on src/main/java/com/photostudio/web/servlet/order/AllOrdersServlet.java:
```diff
@@ -32,7 +35,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) {
         } catch (IOException e) {
-            throw new RuntimeException("AllOrdersServlet error", e);
+            LOG.error("AllOrdersServlet error", e);
+            throw new RuntimeException("AllOrdersServlet error");
```
throw new RuntimeException("AllOrdersServlet error", **e**);

OleksandrGubariev committed:
> 1. Add exception e to catch block
> 2. DefaultUserService:change debug to info
> 3. DefaultSecurityService: change message password to Login/password
> 4. Add to Servlets info logs

OleksandrGubariev committed:
Merge branch 'develop' into lg-1-Add-file-appender
